documentation/binding-data/tutorial-flow-data-provider.asciidoc

```diff
@@ -53,535 +112,429 @@ grid.setItems(
 
+For example, you can programmatically select the next item in a Grid, if a current value is selected and there is a next item after it.
 [source, java]
 ----
-DataProvider<Person, String> dataProvider =
-        getDataProvider(service);
-ComboBox<Person> comboBox = new ComboBox<>();
-comboBox.setDataProvider(dataProvider);
-----
+List<Person> allPersons = repo.findAll();
+GridListDataView<Person> gridDataView = grid.setItems(allPersons);
 
-==== Filtering in the Grid Component
+Button selectNext = new Button("Next", e -> {
+    grid.asSingleSelect().getOptionalValue().ifPresent(p -> {
+        gridDataView.getNextItem(p).ifPresent(
+                next -> grid.select(next)
+        );
+    });
+});
+----
 
-You can use the `withConfigurableFilter` method on a data provider to create a data provider wrapper that allows you to configure the filter that is passed through the query. 
+== Updating the Shown Data
 
-All components that use the same data provider refresh their data when a new filter is set.
+A typical scenario in Vaadin apps is that data displayed in, for example, a `Grid` component, is edited elsewhere in the application.
+Editing the item elsewhere does not automatically update the UI in a listing component.
+An easy way to refresh the component's content is to call `setItems()` again with the fresh data.
+Alternatively, you can use more fine-grained APIs in the `DataView` to update just a portion of the dataset.
 
-*Example*: Using the `withConfigurableFilter` method to create a data provider wrapper.
+For example, you can modify a displayed item and notify `Grid` about the updates to the item through the `DataView` API.
 
 [source, java]
 ----
-DataProvider<Employee, String> employeeProvider =
-        getEmployeeProvider();
+Person person = new Person();
+person.setName("Jorma");
+person.setEmail("old@gmail.com");
 
-ConfigurableFilterDataProvider<Employee, Void, String>
-    wrapper = employeeProvider.withConfigurableFilter();
+GridListDataView<Person> gridDataView = grid.setItems(person);
 
-Grid<Employee> grid = new Grid<>();
-grid.setDataProvider(wrapper);
-grid.addColumn(Employee::getName).setHeader("Name");
+Button modify = new Button("Modify data", e -> {
+    person.setEmail("new@gmail.com");
 
-searchField.addValueChangeListener(event -> {
-    String filter = event.getValue();
-    if (filter.trim().isEmpty()) {
-        // null disables filtering
-        filter = null;
-    }
-
-    wrapper.setFilter(filter);
+    // The component shows the old email until notified of changes
+    gridDataView.refreshItem(person);
 });
 ----
 
-* The filter type of the `wrapper` instance is `Void`. This means that the data provider does not support further filtering through the query. It is therefore not possible to use this data provider with a combo box.
-
-=== Refreshing Data from a Backend Service
+If you are mutating the data set bound to a component like grid, you need to reset the data binding.
```

Comment:
Not sure what is the exact meaning of this. But, apparently, the meaning is to use `setItems` to bind a new (mutated) data set.

Reply:
There is a similar comment about this.

The exact method `setItems`   should be added here because `reset`  means for me something else.
It's complete replacement the whole data. Not just a reset.

I would like to see some explanation what should I do (reset) if `data is mutated` : not the list is mutated.
So I have a list of beans then I've changed one field of the bean. List is not changed. What should I do to see the changes in the component ?

Reply:
I'm now confused with this sentence. You are either call the `refreshItem` after making a change in the item's field, or you use a mutation methods like `add/remove/setFilter` and then the `refreshAll` (`reset`) is invoked automatically. So, basically, the developer does not have to `reset` the data. Don't know what Matti wanted to emphasise here, but I'd rather remove this sentence.